framework_agreement/model/pricelist.py

```diff
@@ -16,75 +14,419 @@
 #
 #    You should have received a copy of the GNU Affero General Public License
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
-from datetime import datetime
-from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
-from openerp.osv import orm, fields
+import time
+from openerp import models, fields, api, tools, exceptions, osv
+from openerp.tools.translate import _
 
 
-# Using new API seem to have side effect on
-# other official addons
-class product_pricelist(orm.Model):
+class Pricelist(models.Model):
+    _inherit = "product.pricelist"
 
-    """Add framework agreement behavior on pricelist"""
+    portfolio_id = fields.Many2one(
+        'framework.agreement.portfolio',
+        'Portfolio',
+    )
 
-    _inherit = "product.pricelist"
+    picking_type_id = fields.Many2one('stock.picking.type', 'Deliver to')
+    payment_term_id = fields.Many2one('account.payment.term', 'Payment terms')
+    terms_of_payment = fields.Char()
+    incoterm_id = fields.Many2one(
+        'stock.incoterms',
+        'Incoterm',
+        help="International Commercial Terms are a series of predefined "
+        "commercial terms used in international transactions.")
+    origin_address_id = fields.Many2one('res.partner', 'Origin Address')
```

Comment:
If this changes it is needed to adapt https://github.com/OCA/vertical-ngo/pull/187